core/src/main/java/com/netease/arctic/table/TableIdentifier.java

```diff
@@ -38,15 +38,43 @@ public TableIdentifier() {
   }
 
   private TableIdentifier(String catalog, String database, String tableName) {
-    this.catalog = Preconditions.checkNotNull(catalog, "Catalog name must not be null.");
```

Comment:
I cannot see the reason why we should remove the check here.

Reply:
because catalog and db can be null in some case, You can check outside the method